### PR TITLE
Performance Improvements to the Codebase.

### PR DESCRIPTION
**Pull request by bordoni**
There are PR's in Pro, Common, TEC, ET and VE related to these changes.

The changes below yielded an improvement of between 10% and up to 25% on certain pages, specially around the one using Widgets.

https://github.com/the-events-calendar/events-virtual/pull/370
https://github.com/the-events-calendar/event-tickets/pull/2275
https://github.com/the-events-calendar/events-pro/pull/1889
https://github.com/the-events-calendar/tribe-common/pull/1708 
https://github.com/the-events-calendar/the-events-calendar/pull/3763

### CODE REVIEW

**Comment by jesseeproductions** [comment]
The changes look good. Not sure if the tests failing here are because of the changes or from something else.

File: tests/wpunit/Tribe/Events/ORM/Events/CreateTest.php [hunk 1]
```diff
@@ -16,6 +16,14 @@ public function setUp() {
 		$this->factory()->organizer = new Organizer();
 		// To support taxonomy term creation and assignment.
 		wp_set_current_user( $this->factory()->user->create( [ 'role' => 'administrator' ] ) );
+
+		if ( isset( tribe( 'cache' )['option_gmt_offset'] ) ) {
```
Comment (Camwyn) [inline tests/wpunit/Tribe/Events/ORM/Events/CreateTest.php:20]: We do this _so.many.times_ - can we use a function or filter for this? (not pressing)

**Review by Camwyn — DISMISSED** [review]
Looks good - one idea for future enhancement.

File: tests/wpunit/Tribe/Events/ORM/Events/CreateTest.php (hunk 1 above)
Comment (bordoni) [reply tests/wpunit/Tribe/Events/ORM/Events/CreateTest.php]: That is an amazing suggestion i added `tribe( 'cache' )->reset();`

**Pull request merged**